api-participant/src/main/java/bio/terra/pearl/api/participant/controller/registration/RegistrationController.java

```diff
@@ -42,11 +41,13 @@ public RegistrationController(
   public ResponseEntity<Object> register(
       String portalShortcode, String envName, UUID preRegResponseId, RegistrationInfo body) {
     var environmentName = EnvironmentName.valueOfCaseInsensitive(envName);
+    // TODO: seems like user should be authorized (tokenLogin) before registering -DC
```

Comment:
there are some registration flows where the registration process is what creates the user, so we can't require the user (See lines 101-107 of RegistrationService).  What we should do though, is move the `requireToken` check below to the first line of this method, and then use the username from the token for registration, rather than the username from the body

Reply:
Sure, I can do that, but it seems like this code, along with RegistrationService.register could use some reorganization? 
Not investigating too deeply, it feels like new user vs existing user could be sorted out at this level to make the service code cleaner. But my original concern here is the fate of the registration data if `tokenLogin` fails. Does it get stranded?
Also, we are using email from the request body, but not verifying it against the email in the token claim?

Reply:
Yeah, I think we'll learn more from HeartHive about how this is going to flow, and might want to restructure, but that's beyond the scope of this PR.  

Yes, there are some corner cases where registration data gets stranded -- since they complete the registration data before they sign in, with certain pernicious patterns of browser refresh/internet connectivity problems, we can lose the user->registration connection.  We attempt to mitigate this with local storage and browser retries, but there's only so much we can do since they completed the form prior to creating an account

Yes, it looks like this is using email from request body without validating against the token claim, which we should fix.  (either in this or in a separate ticket)

Reply:
I'd like to leave the email validation for a separate ticket where we clean up this code, if that's okay with you. 
By separating the two cases the logic will be cleaner, since as things stand now we do not have the email from the token until after registration.